retesteth/Socket.cpp

```diff
@@ -136,20 +164,28 @@ namespace
         if (curl)
         {
             std::unique_ptr<std::string> httpData(new std::string());
+            struct WriteThis wt;
+            wt.readptr = _req.c_str();
+            wt.sizeleft = _req.size();
+
             curl_easy_setopt(curl, CURLOPT_URL, url.c_str());
             curl_easy_setopt(curl, CURLOPT_BUFFERSIZE, 3000000);
             curl_easy_setopt(curl, CURLOPT_WRITEFUNCTION, writecallback);
             curl_easy_setopt(curl, CURLOPT_WRITEDATA, httpData.get());
+            curl_easy_setopt(curl, CURLOPT_READFUNCTION, readcallback);
+            curl_easy_setopt(curl, CURLOPT_READDATA, &wt);
+            curl_easy_setopt(curl, CURLOPT_POST, 1L);
+
             struct curl_slist *header = NULL;
+            header = curl_slist_append(header, "Accept: text/plain");
```

Comment:
Should this accept applicaiton/json instead?  Or both?  This currently breaks us.